ingestion-edge/tests/integration/conftest.py

```diff
@@ -100,8 +102,14 @@ def server(
             assert process.poll() is None  # server still running
             yield "http://localhost:%d" % ports.pop()
         finally:
-            process.kill()
-            process.wait()
+            try:
+                # allow one second for graceful termination
+                process.terminate()
```

Comment:
this allows `--cov` to pick up coverage from integration tests when `--server` isn't specified